addons/account/models/account_move.py

```diff
@@ -1726,6 +1727,17 @@ def _inverse_name(self):
             move.name and move.name != '/'
         ))
 
+    def _inverse_fiscal_position_id(self):
+        # _origin is not working in inverse so we need fatched data from database to check old value.
+        self.env.cr.execute("SELECT id,fiscal_position_id FROM account_move WHERE id in %s", [tuple(self.ids)])
+        old_fiscal_position_id = dict(self.env.cr.fetchall())
+        for move in self:
+            if old_fiscal_position_id.get(move.id) == move.fiscal_position_id.id:
+                continue
+            if (old_fiscal_position_id.get(move.id) or move.fiscal_position_id) and move.state == 'draft':
+                move.line_ids._compute_tax_ids()
+                move.line_ids._compute_account_id()
```

Comment:
@william-andre `_compute_tax_ids` and `_compute_account_id` both auto map tax and accounts
mapping is based on product configuration so we can't map directly what exists in account move line

Reply:
@william-andre Then It's not useful.
because for example, I am applying for an inter-state fiscal position then it converts `GST` tax to `IGST`
Now I apply intra-state and we do not have mapping for `IGST` to `GST` because it's set on a product.

this must behave the same as a Sale else no meaning of this task or PR.

Reply:
Can you show me where it is set on the product?
Fiscal positions shouldn't have anything to do with the products.

Reply:
no, I am talking about `GST` tax set on products, not Fiscal positions.
so no mapping in Fiscal positions from `IGST` to `GST`

Reply:
> Yes, but we don't want to recompute based on the product/partner.
simple, then it works the same as `Sale`?

Reply:
> no, I am talking about `GST` tax set on products, not Fiscal positions. so no mapping in Fiscal positions from `IGST` to `GST`

But the IGST is already set by either the product or the user. So the mapping should now make it GST.

Reply:
`map_accounts` and `map_tax` don't have any information about the product...

Reply:
> But the IGST is already set by either the product or the user. So the mapping should now make it GST.
Normally we do not write mapping on Fiscal positions that we expect the user needs to set on a product.
In Indian, it's GST tax that we expect users to set on products.
if not then it's wrong configuration

Reply:
> `map_accounts` and `map_tax` don't have any information about the product...
obviously, but we write a map based on what we set on a product.